### PR TITLE
[utils] include fmt/xchar.h

### LINKED ISSUE #19897: Kodi 19.1 does not compile against libfmt 8.0.0
### Describe the bug
Here is a clear and concise description of what the problem is:

Kodi 19.1 does not compile against libmt 8.0.0.

Compilation fails with this output:
```
In file included from /var/tmp/portage/media-tv/kodi-19.1/work/xbmc-19.1/xbmc/utils/HTMLUtil.cpp:11:
/var/tmp/portage/media-tv/kodi-19.1/work/xbmc-19.1/xbmc/utils/StringUtils.h: In instantiation of ‘static std::wstring StringUtils::Format(const wstring&, Args&& ...) [with Args = {const wchar_t*}; std::wstring = std::__cxx11::basic_string<wchar_t>]’:
/var/tmp/portage/media-tv/kodi-19.1/work/xbmc-19.1/xbmc/utils/HTMLUtil.cpp:221:55:   required from here
/var/tmp/portage/media-tv/kodi-19.1/work/xbmc-19.1/xbmc/utils/StringUtils.h:98:32: error: no matching function for call to ‘format(const wstring&, const wchar_t*&)’
   98 |     auto result = ::fmt::format(fmt, EnumToInt(std::forward<Args>(args))...);
      |                   ~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/fmt/format.h:44,
                 from /var/tmp/portage/media-tv/kodi-19.1/work/xbmc-19.1/xbmc/utils/StringUtils.h:34,
                 from /var/tmp/portage/media-tv/kodi-19.1/work/xbmc-19.1/xbmc/utils/HTMLUtil.cpp:11:
/usr/include/fmt/core.h:2889:17: note: candidate: ‘std::string fmt::v7::format(fmt::v7::format_string<T ...>, T&& ...) [with T = {const wchar_t*&}; std::string = std::__cxx11::basic_string<char>; fmt::v7::format_string<T ...> = fmt::v7::basic_format_string<char, const wchar_t*&>]’
 2889 | FMT_INLINE auto format(format_string<T...> fmt, T&&... args) -> std::string {
      |                 ^~~~~~
/usr/include/fmt/core.h:2889:44: note:   no known conversion for argument 1 from ‘const wstring’ {aka ‘const std::__cxx11::basic_string<wchar_t>’} to ‘fmt::v7::format_string<const wchar_t*&>’ {aka ‘fmt::v7::basic_format_string<char, const wchar_t*&>’}
 2889 | FMT_INLINE auto format(format_string<T...> fmt, T&&... args) -> std::string {
      |                        ~~~~~~~~~~~~~~~~~~~~^~~
In file included from /var/tmp/portage/media-tv/kodi-19.1/work/xbmc-19.1/xbmc/utils/StringUtils.h:34,
                 from /var/tmp/portage/media-tv/kodi-19.1/work/xbmc-19.1/xbmc/utils/HTMLUtil.cpp:11:
/usr/include/fmt/format.h:2781:13: note: candidate: ‘template<class Locale, class ... T, typename std::enable_if<fmt::v7::detail::is_locale<Locale>::value, int>::type <anonymous> > std::string fmt::v7::format(const Locale&, fmt::v7::format_string<T ...>, T&& ...)’
 2781 | inline auto format(const Locale& loc, format_string<T...> fmt, T&&... args)
      |             ^~~~~~
/usr/include/fmt/format.h:2781:13: note:   template argument deduction/substitution failed:
In file included from /usr/include/fmt/format.h:44,
                 from /var/tmp/portage/media-tv/kodi-19.1/work/xbmc-19.1/xbmc/utils/StringUtils.h:34,
                 from /var/tmp/portage/media-tv/kodi-19.1/work/xbmc-19.1/xbmc/utils/HTMLUtil.cpp:11:
/usr/include/fmt/format.h:2780:11: error: no type named ‘type’ in ‘struct std::enable_if<false, int>’
 2780 |           FMT_ENABLE_IF(detail::is_locale<Locale>::value)>
      |           ^~~~~~~~~~~~~
```

Originally reported at https://bugs.gentoo.org/797619

## Expected Behavior
Here is a clear and concise description of what was expected to happen:

Kodi should be able to compile against the version 8.0.0 of fmt

## Actual Behavior

Compilation fails.

### To Reproduce

1. Install fmt 6.2.0
2. Compile Kodi

### Debuglog

The debuglog can be found here:

n/a, this is a compilation problem, not a runtime one.

### Screenshots 
Here are some links or screenshots to help explain the problem:

n/a, this is a compilation problem, not a runtime one.

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:

n/a, this is a compilation problem, not a runtime one.

### Your Environment
Used Operating system:

 - [ ] Android
 - [ ] iOS
 - [ ] tvOS
 - [X] Linux
 - [ ] OSX
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name: Gentoo/Linux
 - Kodi version: 19.1

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

### PR DESCRIPTION
**Pull request by candrews**
## Description
fmt 8 moved wchar/custom char overloads to xchar.h, so for fmt 8 compatibility, xchar.h must be included

See https://github.com/fmtlib/fmt/commit/76ee490468212f8705a1421b4c88f8f814b2d351

Fixes #19897

## Motivation and context

Provides compatibility with fmt 8.

## How has this been tested?

Kodi now compiles :)

## What is the effect on users?

Kodi now compiles against fmt 8.0.0.

## Screenshots (if appropriate):
n/a

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [X] **None of the above** (please explain below)

Compatibility with the latest version of a dependency.

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [X] All new and existing tests passed

### CODE REVIEW

File: xbmc/utils/StringUtils.h [hunk 1]
```diff
@@ -31,11 +31,12 @@
 #undef FMT_DEPRECATED
 #define FMT_DEPRECATED
 #endif
-#include <fmt/format.h>
-
 #include "XBDateTime.h"
 #include "utils/params_check_macros.h"
 
+#include <fmt/format.h>
+#include <fmt/xchar.h>
```
Comment (Paxxi) [inline xbmc/utils/StringUtils.h:38]: It looks like `xchar.h` is new in v8 so we need a compatibility check like this.
```suggestion
#if FMT_VERSION >= 80000
#include <fmt/xchar.h>
#endif
```

**Review by Paxxi — CHANGES_REQUESTED** [review]
Thank you for fixing this. I think with the suggested change or something similar this should be good to go.

File: xbmc/utils/StringUtils.h (hunk 1 above)
Comment (candrews) [reply xbmc/utils/StringUtils.h:38]: Thank you!

**Review by Paxxi — APPROVED** [review]

**Comment by wsnipex** [comment]
I guess this should have been merged before the v19 backport

**Pull request merged**